applications/MappingApplication/custom_mappers/nearest_neighbor_mapper.h

```diff
@@ -22,39 +22,126 @@
 
 // Project includes
 #include "mapper.h"
+#include "custom_utilities/mapping_matrix_builder.h"
+#include "custom_searching/interface_communicator.h"
+#include "custom_utilities/interface_preprocessor.h"
+#include "custom_utilities/mapper_flags.h"
+#include "custom_utilities/mapper_local_system.h"
 
 
 namespace Kratos
 {
 
-///@name Kratos Globals
+///@name Kratos Classes
 ///@{
 
-///@}
-///@name Type Definitions
-///@{
+class NearestNeighborInterfaceInfo : public MapperInterfaceInfo
+{
+public:
 
-///@}
-///@name  Enum's
-///@{
+    /// Default constructor.
+    NearestNeighborInterfaceInfo() {}
 
-///@}
-///@name  Functions
-///@{
+    NearestNeighborInterfaceInfo(const CoordinatesArrayType& rCoordinates,
+                                 const IndexType SourceLocalSystemIndex,
+                                 const IndexType SourceRank)
+        : MapperInterfaceInfo(rCoordinates, SourceLocalSystemIndex, SourceRank) {}
+
+    MapperInterfaceInfo::Pointer Create() const override
+    {
+        return Kratos::make_shared<NearestNeighborInterfaceInfo>();
+    }
+
+    MapperInterfaceInfo::Pointer Create(const CoordinatesArrayType& rCoordinates,
+                                        const IndexType SourceLocalSystemIndex,
+                                        const IndexType SourceRank=0) const override
+    {
+        return Kratos::make_shared<NearestNeighborInterfaceInfo>(
+            rCoordinates,
+            SourceLocalSystemIndex,
+            SourceRank);
+    }
+
+    void ProcessSearchResult(const InterfaceObject::Pointer& rpInterfaceObject,
+                             const double NeighborDistance) override;
+
+    void GetValue(int& rValue,
+                  const InfoType ValueType=MapperInterfaceInfo::InfoType::Dummy) const override
+    {
+        rValue = mNearestNeighborId;
+    }
+
+    void GetValue(double& rValue,
+                  const InfoType ValueType=MapperInterfaceInfo::InfoType::Dummy) const override
+    {
+        rValue = mNearestNeighborDistance;
+    }
+
+private:
+
+    int mNearestNeighborId;
```

Comment:
Uninitialized variable
```suggestion
    int mNearestNeighborId = -1;
```